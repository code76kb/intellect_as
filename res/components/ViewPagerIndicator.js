import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from 'prop-types'

export default function ViewPagerIndicator(props) {

    let pageArray = Array.from({ length: props.pageCount });
    return (
      <View style={styles.indicatorContainer}>
        {pageArray.map((v, i) => {
          return <View key={`indi_${i}`} style={[styles.indicatorBar, i > 0 ? { marginLeft: 10 } : null, i > props.currentPage ? { backgroundColor: props.color } : {backgroundColor: props.activeColor } ]} />
        })}
      </View>
    );
}


const styles = StyleSheet.create({
    indicatorContainer: { 
        height: 10, 
        flexDirection: 'row', 
        alignItems: 'center', 
        paddingHorizontal: 10 
    },
    indicatorBar: {
         flex: 1, 
         height: 4, 
         borderRadius: 4
    },
});

ViewPagerIndicator.propTypes = {
    pageCount: PropTypes.number.isRequired,
    currentPage: PropTypes.number.isRequired,
    color:PropTypes.string,
    activeColor:PropTypes.string
}

ViewPagerIndicator.defaultProps = {
    pageCount: 0,
    currentPage: 0,
    color:"#819da7",
    activeColor:'white'
}
