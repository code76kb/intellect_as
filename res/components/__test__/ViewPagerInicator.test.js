import React from 'react';
import renderer from 'react-test-renderer';
import ViewPagerIndicator from '../ViewPagerIndicator' 
it('Render ViewPagerIndicator  Comp :', () => {
    const tree = renderer.create(
        <ViewPagerIndicator pageCount={3} currentPage={1} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})