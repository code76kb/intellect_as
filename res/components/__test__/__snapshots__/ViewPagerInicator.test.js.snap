// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ViewPagerIndicator  Comp : 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 4,
          "flex": 1,
          "height": 4,
        },
        null,
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 4,
          "flex": 1,
          "height": 4,
        },
        Object {
          "marginLeft": 10,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 4,
          "flex": 1,
          "height": 4,
        },
        Object {
          "marginLeft": 10,
        },
        Object {
          "backgroundColor": "#819da7",
        },
      ]
    }
  />
</View>
`;
